tests/test_warpedvrt.py

```diff
@@ -37,15 +49,46 @@ def test_warped_vrt(path_rgb_byte_tif):
 def test_warped_vrt_source(path_rgb_byte_tif):
     """A VirtualVRT has the expected source dataset."""
     with rasterio.open(path_rgb_byte_tif) as src:
-        vrt = WarpedVRT(src, dst_crs='EPSG:3857')
+        vrt = WarpedVRT(src, dst_crs=DST_CRS)
         assert vrt.src_dataset == src
 
 
+def test_warped_vrt_set_src_crs(path_rgb_byte_tif, tmpdir):
+    """A VirtualVRT without CRS works only with parameter src_crs"""
+    path_crs_unset = str(tmpdir.join('rgb_byte_crs_unset.tif'))
+    _copy_update_profile(path_rgb_byte_tif, path_crs_unset, crs=None)
+    with rasterio.open(path_rgb_byte_tif) as src:
+        original_crs = src.crs
+    with rasterio.open(path_crs_unset) as src:
+        with pytest.raises(Exception):
+            with WarpedVRT(src, dst_crs=DST_CRS) as vrt:
+                pass
+        with WarpedVRT(src, src_crs=original_crs, dst_crs=DST_CRS) as vrt:
+            assert vrt.src_crs == original_crs
+
+
+def test_warped_vrt_set_src_transform(path_rgb_byte_tif, tmpdir):
+    """A VirtualVRT without transform works only with parameter src_transform"""
+    path_transform_unset = str(tmpdir.join('rgb_byte_transform_unset.tif'))
+    _copy_update_profile(path_rgb_byte_tif, path_transform_unset, transform=None)
+    with rasterio.open(path_rgb_byte_tif) as src:
+        original_transform = src.transform
+    with rasterio.open(path_transform_unset) as src:
+        with pytest.raises(Exception):
+            with WarpedVRT(src, dst_crs=DST_CRS) as vrt:
+                pass
+        with WarpedVRT(
+                src,
+                src_transform=original_transform,
+                dst_transform=original_transform) as vrt:
+            assert vrt.src_transform == original_transform
```

Comment:
@j08lue is the intent of this to create a warped VRT that represents a no-op? I don't think that's a valid use case, though I believe we should fail more clearly in that case. Would you be willing to remove this test? I'm fine with letting this remain a bug to be fixed after 1.0.

Reply:
Yes, let's remove this one test. I think there's a problem lower down, maybe even in GDAL, with the identical src and dst transforms.